### DISCUSSION TITLE
Multiple translations - choose language through the API

### DISCUSSION THREAD

**Discussion by danieljwestman** [directus/directus#10362] [Archive]
Directus is handling multi languages and translations great within the app.

With the API you can get an items translations with deep query (?fields=translations.*).

But; how do you choose which language to get in the response? I can't find any info on this, but I'm sure I'm missing something obvious...?

Changing language on the current user doesn't do any difference.

**Comment by rijkvanzanten** [comment]
A translations field is no different from a regular many to many field. Therefore, the same rules for nested relational filters apply, for example:

```
?deep[translations][_filter][language_code][_eq]=nl-NL
```

will only nest the Dutch translations 🙂 

That being said, there's an open discussion going on in https://github.com/directus/directus/discussions/6654 to add a dedicated `lang` query parameter that would effectively function as a shorthand for the above filter

**Reply by vitalijalbu** [reply]
what's the correct collection to call on URL API?
**categories** or **categories_translations**

**Comment by danieljwestman** [comment]
> A translations field is no different from a regular many to many field. Therefore, the same rules for nested relational filters apply
Ah, of course. I'm hip.

> That being said, there's an open discussion going on in #6654 to add a dedicated lang query parameter that would effectively function as a shorthand for the above filter

Yeah, `?lang=X` would be a great enhancement. Translations is an important part and should be elevated in the API request spec. In our particular case 95% of the fields in the item are translated into more than 3 languages. Working with translations in the app is a pleasure, especially with the "new" side-by-side functionality 👍🏼

**Comment by chrislsww** [comment]
Just wanted to add that if you not only want the translations ID to be displayed in the response, you also need to attach the fields in the request:

`{{baseUrl}}/items/articles?deep[translations][_filter][languages_code][_eq]=en-US&fields=id,status,date_created,image,date_published,translations.title,translations.slug,translations.content`

That would result in:

```
{
    "data": [
        {
            "id": "6bece7ab-4cd8-4b2b-b0e6-3916ff5af501",
            "status": "published",
            "date_created": "2025-03-15T09:01:03.905Z",
            "image": "39d7ba7d-324b-4eef-bae5-36620af4862d",
            "date_published": "2025-03-14T11:00:00.000Z",
            "translations": [
                {
                    "title": "First Post",
                    "slug": "first-post",
                    "content": "<p>Here is the first article.</p>"
                }
            ]
        },
        {
            "id": "d435c0a4-ea8b-4c66-b023-8c8713ea043c",
            "status": "published",
            "date_created": "2025-03-15T12:12:47.138Z",
            "image": null,
            "date_published": "2025-03-13T11:00:00.000Z",
            "translations": [
                {
                    "title": "My second article",
                    "slug": "my-second-article",
                    "content": "<p>Here is the second article</p>"
                }
            ]
        }
    ]
}
```
